src/components/button-toggle/button-toggle.pw.tsx

```diff
@@ -730,23 +729,24 @@ test.describe("Event tests for group component", () => {
     expect(callbackCount).toBe(0);
   });
 
-  test("should call onChange callback with undefined when a click event is triggered on the currently-selected button and the allowDeselect prop is true", async ({
+  test("calls onChange callback with an undefined value, when the currently-selected button is clicked and the allowDeselect prop is true", async ({
     mount,
     page,
   }) => {
-    const callbackStub = sinon.stub();
+    let onChangeValue: unknown;
     await mount(
       <ButtonToggleGroupComponent
         value="foo"
         allowDeselect
-        onChange={callbackStub}
+        onChange={(event, value) => {
+          onChangeValue = value;
+        }}
       />
     );
 
-    await buttonTogglePreview(page).nth(0).click();
+    await page.getByRole("button", { name: "foo" }).click();
 
-    const onChangeCallArgs = callbackStub.getCall(0).args[1];
-    expect(onChangeCallArgs).toEqual(undefined);
+    await expect.poll(() => onChangeValue).toBeUndefined();
```

Comment:
question (non-blocking): this seemed to work previously with a synchronous assertion, do we need to use `poll` here, it seems to be for more complex scenarios like API requests?

Reply:
Good point. I made this assertion asynchronous to account for potential latency in the execution of `onChange`. However, since it's a synchronous operation, I agree it's unnecessary here.